packages/utils/src/types.ts

```diff
@@ -282,8 +276,17 @@ export type RegistryWidgetsType<T = any, S extends StrictRJSFSchema = RJSFSchema
   [name: string]: Widget<T, S, F>;
 };
 
+export type RJSFBaseProps<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any> = {
+  /** The schema object for the field being described */
+  schema: S;
+  /** The uiSchema object for this description field */
+  uiSchema?: UiSchema<T, S, F>;
+  /** The `registry` object */
+  registry: Registry<T, S, F>;
+};
+
 /** The set of RJSF templates that can be overridden by themes or users */
```

Comment:
Consider adding documentation around the change you made below, here as well

Reply:
I'm not sure what kind of documentation I should add, could you specify please ?

Reply:
I suggest mentioning the ability to add per-field customizations and maybe add an example of a custom template since it is now supported?

Reply:
Wow, I'm super glad this feature is finally getting into the master branch. It's almost having an one year birthday 🎂.  We use this feature extensively to customize the layout of the form. I think its a very nice use case to describe in the docs, since I'm aware it's a very long desired flexibility. I could also contribute by providing the Layout template.